### ISSUE TITLE
[BUG][Accessibility] Autocomplete element has missing semantics

### ISSUE THREAD

**Issue by jamiegaskell** [scottish-government-design-system/design-system#119]
## Description of the issue

The [autocomplete](https://designsystem.gov.scot/components/autocomplete) (combobox) element has missing ARIA and semantics. This includes:

### Input element

* The `<input>` element does not use `role="combobox"`.
* The expandable state is not determined using `aria-expanded`.
* The `<input>` element does not use the attribute `aria-controls` to identify the listbox element that serves as the popup. This should reference the `id` of the listbox.

![The listbox element](https://github.com/user-attachments/assets/3ffebcef-5fde-4c76-8a2c-72f4de618a91)

![DOM, the input element is missing ARIA attributes](https://github.com/user-attachments/assets/bd6f62aa-9bdf-47a2-967c-0917f345d403)

### Listbox element

* Selected options are not determined using `aria-selected`.

![DOM, selected options are not determined using aria-selected](https://github.com/user-attachments/assets/c57f5566-45af-4962-a6e9-6d8a157064ab)

## Steps to reproduce the issue

1. Navigate onto the element and observe that information such as the expanded state and selected options are not announced
2. Inspect the code and compare with examples from the ARIA authoring practices page

## Expected behaviour

Although not finalised, the ARIA authoring practices provide a good working example. See [Editable Combobox With List Autocomplete](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/)

## Environment

- Operating system: Windows 10
- Browser: Edge
- Assistive technology: JAWS 2023

**Comment by jsutcliffe** [comment]
This is a complex topic without a good 'official' suggestion from W3C. An autocomplete requesting results dynamically is a different scenario from the fixed list of options in the W3C example you link to.

## `role=combobox`:
We do not use the combobox role on the input element because it is not supported on inputs of type "search".

## `aria-expanded`:
Like the combobox role, `aria-expanded` is not supported on search inputs. We actually did have those attributes on a previous iteration of the component but dropped them when we decided that using the correct input type was more important.

In testing, I found that the expanded/collapsed state of the suggestions list was not being announced consistently with when we used `aria-expanded`. Instead, I've settled for an `aria-live` region that tells the user that there are options available to choose from (e.g. "These are 4 options"). This has worked consistently across devices/browsers I have tested on.

## `aria-owns`/`aria-controls`:
A previous accessibility audit instructed us to use `aria-owns` on the input, but on reflection I agree that `aria-controls` seems more appropriate for this.

## `aria-selected`:
We are using `aria-activedescendant` to flag the selected option. Setting `aria-selected` on the selected option was not getting announced reliably in screen readers, whereas `aria-activedescendant` behaves well in the testing I've carried out.
